joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/AstForStatementsCreator.scala

```diff
@@ -24,13 +25,19 @@ import scala.collection.mutable
 
 trait AstForStatementsCreator(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
 
-  protected def astForBlockStatement(blockStmt: IASTCompoundStatement, order: Int = -1): Ast = {
-    val codeString = code(blockStmt)
-    val blockCode  = if (codeString == "{}" || codeString.isEmpty) Defines.Empty else codeString
-    val node = blockNode(blockStmt, blockCode, registerType(Defines.Void))
+  protected def astForBlockStatement(blockStmt: IASTCompoundStatement, blockNode: NewBlock, order: Int = -1): Ast = {
+    val codeString  = code(blockStmt)
+    val blockCode   = if (codeString == "{}" || codeString.isEmpty) Defines.Empty else codeString
```

Comment:
I see that it's existing behavior, but I don't think this is helpful: there is a difference between a unknown/missing value (`<empty>`), and an actual empty value. So we should just use `codeString` here universally.